api/src/main/java/com/graphhopper/util/PMap.java

```diff
@@ -83,145 +80,93 @@ public static PMap read(String[] args) {
-        if (!Helper.isEmpty(str)) {
-            try {
-                return Long.parseLong(str);
-            } catch (Exception ex) {
-            }
-        }
-        return _default;
+    public boolean getBool(String key, boolean _default) {
+        Object object = map.get(key);
+        return object instanceof Boolean ? (Boolean) object : _default;
     }
 
     public int getInt(String key, int _default) {
-        String str = get(key);
-        if (!Helper.isEmpty(str)) {
-            try {
-                return Integer.parseInt(str);
-            } catch (Exception ex) {
-            }
-        }
-        return _default;
+        Object object = map.get(key);
+        return object instanceof Number ? ((Number) object).intValue() : _default;
     }
 
-    public boolean getBool(String key, boolean _default) {
-        String str = get(key);
-        if (!Helper.isEmpty(str)) {
-            try {
-                return Boolean.parseBoolean(str);
-            } catch (Exception ex) {
-            }
-        }
-        return _default;
+    public long getLong(String key, long _default) {
+        Object object = map.get(key);
+        return object instanceof Number ? ((Number) object).longValue() : _default;
     }
 
-    public double getDouble(String key, double _default) {
-        String str = get(key);
-        if (!Helper.isEmpty(str)) {
-            try {
-                return Double.parseDouble(str);
-            } catch (Exception ex) {
-            }
-        }
-        return _default;
+    public float getFloat(String key, float _default) {
+        Object object = map.get(key);
+        return object instanceof Number ? ((Number) object).floatValue() : _default;
     }
 
-    public float getFloat(String key, float _default) {
-        String str = get(key);
-        if (!Helper.isEmpty(str)) {
-            try {
-                return Float.parseFloat(str);
-            } catch (Exception ex) {
-            }
-        }
-        return _default;
+    public double getDouble(String key, double _default) {
+        Object object = map.get(key);
+        return object instanceof Number ? ((Number) object).doubleValue() : _default;
     }
 
     public String get(String key, String _default) {
-        String str = get(key);
-        if (Helper.isEmpty(str))
-            return _default;
-
-        return str;
+        Object object = map.get(key);
+        return object instanceof String ? (String) object : _default;
     }
 
-    String get(String key) {
-        if (Helper.isEmpty(key))
-            return "";
-
-        // query accepts camelCase and under_score
-        String val = map.get(Helper.camelCaseToUnderScore(key));
-        if (val == null)
-            return "";
+    public Object getObject(String key, Object _default) {
+        Object object = map.get(key);
+        return object == null ? _default : object;
+    }
 
-        return val;
+    public PMap putObject(String key, Object object) {
```

Comment:
Why not just name this put as well (overload the existing put method)? Because of the camelcase/underscore conversion?